sklearn/multiclass.py

```diff
@@ -770,7 +770,14 @@ def predict(self, X):
         """
         Y = self.decision_function(X)
         if self.n_classes_ == 2:
-            return self.classes_[(Y > 0).astype(int)]
+            if hasattr(self.estimators_[0], "decision_function") and is_classifier(
+                self.estimators_[0]
+            ):
+                thresh = 0
+            else:
+                # predict_proba threshold
+                thresh = 0.5
+            return self.classes_[(Y > thresh).astype(int)]
```

Comment:
Is it worth wrapping this and the lines you mentioned, i.e.: 

https://github.com/scikit-learn/scikit-learn/blob/70eebb992e8a799cbc3d7f1fbfddd104c0908c66/sklearn/multiclass.py#L429-L434

in a private `_threshold` property?

I think this can make the logic explicit and ease maintainability.

Reply:
`OneVsOneClassifier` and `OneVsRestClassifier` do not share a common base class where a `_threshold` property makes sense. Also the threshold is only used in the binary case.

Inspired by your suggestion, I think a helper function makes the logic better [`7372cee` (#22604)](https://github.com/scikit-learn/scikit-learn/pull/22604/commits/7372ceeec6b585321451b4b237735a541af3fd8d)